src/Type/Data/Record.purs

```diff
@@ -0,0 +1,18 @@
+module Type.Data.Record
+  ( class RecordOf
+  ) where
+
+import Type.Row (class RowToList, Cons, Nil, kind RowList)
+
+class RecordOf record fieldType
+instance recordOf
+  :: ( RowToList record fields
+     , FieldOf fields fieldType
+     )
+  => RecordOf (Record record) fieldType
```

Comment:
I think `(Record record)` is the right thing to do here (as opposed to just `record`) but it has the consequence of changing the type of a function beyond simply adding the type constraint.

For example, this:

```purescript
foo :: forall a
     . Record a
    -> Record a
```

becomes:

```purescript
foo :: forall a
     . RecordOf a Boolean
    => a
    -> a
```